bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/driver/Port.java

```diff
@@ -255,6 +265,18 @@ public void modemDBComplete() {
         driver.modemDBComplete(this);
     }
 
+    public void disconnected() {
+        if (isRunning()) {
+            synchronized (disconnected) {
+                if (!disconnected) {
+                    logger.warn("port {} disconnected", logName);
+                    driver.disconnected();
+                    disconnected = true;
+                }
+            }
```

Comment:
You shouldn't synchronize on a non-final field, and you shouldn't synchronize on a boolean. Java synchronization operates against object instances, not against field names, so you would either need to use a dedicated lock object for synchronization:
```
private final Object disconnectLock = new Object();
```
```suggestion
            synchronized (disconnectLock) {
                if (!disconnected) {
                    logger.warn("port {} disconnected", logName);
                    driver.disconnected();
                    disconnected = true;
                }
            }
```

Or you can change your `disconnected` field into one of the atomic utility classes (like `AtomicBoolean`) and then do the following:
```suggestion
            if(!disconnected.getAndSet(true)){
                logger.warn("port {} disconnected", logName);
                driver.disconnected();
            }
```

Reply:
Changed to AtomicBoolean